From 064b74d5e19847794072781405eafe8dfbfca331 Mon Sep 17 00:00:00 2001
From: Vicente Olivert Riera <Vincent.Riera@imgtec.com>
Date: Mon, 29 May 2017 23:24:42 +0300
Subject: [PATCH] Really make -Werror conditional to BUILD_WERROR

Otherwise it will fail with an error message like this one:

elf_getarsym.c:290:9: error: 'n' may be used uninitialized in this
function [-Werror=maybe-uninitialized]
    arsym[n].as_name = NULL;
	 ^
cc1: all warnings being treated as errors

[Vincent: tweak patch for 0.166]
[Bernd: rebased patch for 0.177 & 0.189]
[Dario: make the patch to be applied with fuzz factor 0]

Signed-off-by: "Yann E. MORIN" <yann.morin.1998@free.fr>
Signed-off-by: Vicente Olivert Riera <Vincent.Riera@imgtec.com>
Signed-off-by: Bernd Kuhls <bernd.kuhls@t-online.de>
Signed-off-by: Dario Binacchi <dario.binacchi@amarulasolutions.com>
---
 config/eu.am | 1 -
 1 file changed, 1 deletion(-)

diff --git a/config/eu.am b/config/eu.am
index e6c241f9d88a..cad1baa27a5f 100644
--- a/config/eu.am
+++ b/config/eu.am
@@ -99,7 +99,6 @@ AM_CFLAGS = -std=gnu99 -Wall -Wshadow -Wformat=2 \
 	    $(LOGICAL_OP_WARNING) $(DUPLICATED_COND_WARNING) \
 	    $(NULL_DEREFERENCE_WARNING) $(IMPLICIT_FALLTHROUGH_WARNING) \
 	    $(USE_AFTER_FREE3_WARNING) \
-	    $(if $($(*F)_no_Werror),,-Werror) \
 	    $(if $($(*F)_no_Wunused),,-Wunused -Wextra) \
 	    $(if $($(*F)_no_Wstack_usage),,$(STACK_USAGE_WARNING)) \
 	    $(if $($(*F)_no_Wpacked_not_aligned),$(NO_PACKED_NOT_ALIGNED_WARNING),) \
-- 
2.43.0

